### PR TITLE
Update Logic component in Developer's Guide

### LINKED ISSUE #108: Update Developer's Guide design components
Components to update:

1. Architecture
2. UI component
3. Logic component
4. Model component
5. Storage component
6. Common classes

### PR DESCRIPTION
**Pull request by Avalionnet**
Updated UML diagrams and modified descriptions used in this section.

closes #108

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -80,9 +80,9 @@ The `UI` component,
 **API** :
 [`Logic.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/logic/Logic.java)
```
Comment (JinHao-L) [inline docs/DeveloperGuide.md]: I think the link should be updated to our group repo

**Review by JinHao-L — APPROVED** [review]
Other than the comment above, LGTM 👍

**Review by YangYue128-helen — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Avalionnet) [reply docs/DeveloperGuide.md]: oh yea nice spot 👍🏻

**Pull request merged**